replication/binlogstreamer.go

```diff
@@ -36,6 +36,26 @@ func (s *BinlogStreamer) GetEvent(ctx context.Context) (*BinlogEvent, error) {
 	}
 }
 
+// Get the binlog event with starttime, if current binlog event timestamp smaller than specify starttime
+// return nil event 
+func (s *BinlogStreamer) GetEventWithStartTime(ctx context.Context,startTime time.Time) (*BinlogEvent, error) {
+	if s.err != nil {
+		return nil, ErrNeedSyncAgain
+	}
+	startUnix := startTime.Unix()
+	select {
+	case c := <-s.ch:
```

Comment:
is it better to loop check all the time until the Timestamp is larger than the startUnix, and then return the event?

but this depends on your case, if you think returning nil is fine, I will merge it.

Reply:
Yes, it is suitable for my case . thanks.